IdentityCore/src/util/NSString+MSIDExtensions.h

```diff
@@ -42,26 +46,48 @@
 /*! Encodes the string to pass it as a URL agrument. */
 - (NSString *)msidUrlFormEncode;
 
-/*! Converts base64 String to NSData */
-+ (NSData *)msidBase64UrlDecodeData:(NSString *)encodedString;
-
-/*! Converts NSData to base64 String */
-+ (NSString *)msidBase64UrlEncodeData:(NSData *)data;
-
-- (NSString*)msidComputeSHA256;
-
-/*! Converts string to url */
-- (NSURL *)msidUrl;
-
 /*! Calculates a hash of the passed string. Useful for logging tokens, where we do not log
  the actual contents, but still want to log something that can be correlated. */
 - (NSString *)msidTokenHash;
 
-- (NSOrderedSet<NSString *> *)scopeSet;
+/*! Check if current string is included in the array - case insensitive */
+- (BOOL)msidIsEquivalentWithAnyAlias:(NSArray<NSString *> *)aliases;
+
+/*! Removes a trailing padding for base64url encoding */
+- (NSString *)msidStringByRemovingPadding;
 
+/*!
+ =============================================================================
+ String constructors
+ =============================================================================
+ */
 /*! Generate a URL-safe string of random data */
-+ (NSString *)randomUrlSafeStringOfSize:(NSUInteger)size;
++ (NSString *)msidRandomUrlSafeStringOfByteSize:(NSUInteger)size;
 
-- (BOOL)msidIsEquivalentWithAnyAlias:(NSArray<NSString *> *)aliases;
+/*! Generate a URL-safe string of random data */
++ (NSString *)msidHexStringFromData:(NSData *)data;
+
+/*! Generate a base64 url-encoded string of random data */
++ (NSString *)msidBase64UrlEncodedStringFromData:(NSData *)data;
+
+/*! Generate a www-form-urlencoded string of random data */
++ (NSString *)msidUrlFormEncodedStringFromDictionary:(NSDictionary *)dict;
+
+/*!
+ =============================================================================
+ Convenience methods
+ =============================================================================
+ */
+/*! Convenience method to convert string to url */
++ (NSString *)msidStringFromOrderedSet:(NSOrderedSet *)set;
+
+/*! Convenience method to convert string to url */
+- (NSURL *)msidUrl;
```

Comment:
do we really need this?

Reply:
Moved it to test util, as we have quite a few cases which uses this.